moviepy/video/VideoClip.py

```diff
@@ -721,13 +739,24 @@ def set_make_frame(self, mf):
         self.size = self.get_frame(0).shape[:2][::-1]
 
     @outplace
-    def set_audio(self, audioclip):
-        """Attach an AudioClip to the VideoClip.
-
-        Returns a copy of the VideoClip instance, with the `audio`
+    def set_audio(self, audioclip, start_time=0, keep_original=False):
```

Comment:
few notes

1. In general, would be much better to open independent PRs, one for each proposed new feature. Thiat would be cleaner for the history / release notes and also for the discussion.  
2. the idea of this seems handy but I'm not sure if it's not much "overspecialized" . is it a great advantage comparing with composite the audio clip in advance and then apply to the videoclip? what if instead a "silence" I would like a fadein? or what about if instead to start at a time I would like to end at a time? 

3. I'm sure there is a better way to create a "silence" clip than set `volume 0`. check examples of `make_frame` in the test.  

4. In any case, this feature would needs some tests

Reply:
@Zulko your opinion would be valuable here, specially regarding my comment's point 2 and 3.